hazelcast-it/distribution-it/src/test/java/com/hazelcast/it/hibernate/HibernateIT.java

```diff
@@ -86,6 +86,10 @@ public void testInsertLoad() {
         tx.commit();
         session.close();
 
+        //test can fail on macOS docker-desktop due to small (1-2ms) clock drifts on VM
+        //`HazelcastTimestamper.nextTimestamp` returns future time which leads to cache misses
+        sleepMillis(10);
+
```

Comment:
This is interesting, could you explain more in detail what timestamps are compared?

Would it be possible to wait until some timestamp is later than other, instead of a fixed wait (which may again fail on overloaded jenkins machine..).

Reply:
I've refactor the test to use assertTrueEventually: https://github.com/hazelcast/hazelcast/pull/22947/commits/08a7d3a9af916a7641b3205de50e02c30f0e2f87

We compare timestamp of the cached item against Hibernate transaction timestamp.

This is the place where it occurs: https://github.com/hibernate/hibernate-orm/blob/ab549a48280ead16ae066d269fe4a703496afefb/hibernate-core/src/main/java/org/hibernate/cache/spi/support/AbstractReadWriteAccess.java#L271

Here's `isReadable` method is called: https://github.com/hibernate/hibernate-orm/blob/ab549a48280ead16ae066d269fe4a703496afefb/hibernate-core/src/main/java/org/hibernate/cache/spi/support/AbstractReadWriteAccess.java#L77

It fails only with `HazelcastLocalCacheRegionFactory` because we take cluster time (from docker VM) as the timestamp: https://github.com/hazelcast/hazelcast-hibernate/blob/3f112835fbd077b0c7dc2e7fa1ecbbe065f3f92f/hazelcast-hibernate53/src/main/java/com/hazelcast/hibernate/HazelcastLocalCacheRegionFactory.java#L91

For `HazelcastCacheRegionFactory` we use default implementation: https://github.com/hibernate/hibernate-orm/blob/741b6b71f1552aa224fb61d38b85b32e0b8a19b4/hibernate-core/src/main/java/org/hibernate/cache/spi/AbstractRegionFactory.java#L148 which works fine


Here's the issue from docker-desktop: https://github.com/docker/for-mac/issues/2076. 

I've re-check tests and now it fails because `txTimestamp == item.timestamp` but there's no future time. As I understood from the docker issue above, time drift occurs gradually depending on the uptime and number of mac suspensions